### PR TITLE
chore: update react-native version in example app

### PR DESCRIPTION
**Pull request by esauerbo**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — DISMISSED** [review]

File: packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap [hunk 1]
```diff
@@ -29,7 +29,7 @@ exports[`Carousel calls the orientation handler on orientation change 1`] = `
     removeClippedSubviews={false}
     renderItem={[Function]}
     renderToHardwareTextureAndroid={true}
-    scrollEventThrottle={50}
+    scrollEventThrottle={0.0001}
```
Comment (reesscot) [inline packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap:32]: how did this sneak in here?

Comment (esauerbo) [reply packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap:32]: Not sure why exactly, but the react-native unit tests were failing on snapshots so I had to update them

Comment (calebpollman) [reply packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap:32]: Just an fyi, we shouldn't need to update tests in this PR

File: examples/react-native/package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
     "@xstate/react": "^3.2.2",
     "aws-amplify": "^6.0.6",
     "react": "18.2.0",
-    "react-native": "0.71.14",
+    "react-native": "0.73.4",
```
[the remark this answers is not in the document]
Comment (esauerbo) [reply examples/react-native/package.json:26]: Is 73 not the minor version?

File: packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap (hunk 1 above)
Comment (reesscot) [reply packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap:32]: Maybe tests changed due to bumping from 71 to 73 (which is a major version change in RN land)

File: examples/react-native/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (calebpollman) [reply examples/react-native/package.json:26]: Sorry, reversed the order. In RN, semver minor version is RN major version

File: packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap (hunk 1 above)
Comment (calebpollman) [reply packages/react-native/src/primitives/Carousel/__tests__/__snapshots__/Carousel.spec.tsx.snap:32]: That would be my assumption. RN ships with a jest preset that includes mocked values for native APIs, default value of `scrollEventThrottle` might have changed between major versions

File: yarn.lock [hunk 3]
```diff
@@ -5893,7 +5893,6 @@
     envinfo "^7.7.2"
     execa "^1.0.0"
     hermes-profile-transformer "^0.0.6"
-    ip "^1.1.5"
```
Comment (calebpollman) [inline yarn.lock:5896]: Nice!

Comment (reesscot) [reply yarn.lock:5896]: 💯

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**